### ISSUE TITLE
SideView Generation Hanging with Certain NucleicAcidProfiles

### ISSUE THREAD

**Issue by DNAcer** [404Wolf/NATuG#10]
I'm not sure what's going on, but some combinations of B and T values make the software hang when I try to update graphs. For example, T=1 seems to work with B=8, or 9, or 21, or 19 but not 10 or 13. T=2 works with B=21, or 22, but not 20. 

-Bill

**Comment by 404Wolf** [comment]
@DNAcer 
This error has to do with lines 77-102 of workers\side_view.py, where the NEMids are aligned so that each one has a mate.  Some notes:
- .094 nm is a static value that represents the z distance between a NEMid and its mate. This value should be computed based on the current nucleic_acid_profile, so if you have the formula I can make it dynamic.
- Line 95 is the exact location of the error (you can put a print("Test") there and it will infinitely print test when it's "hanging"). 
I believe the proper solution here is to implement a formula for dynamically generating z_mate.

**Comment by DNAcer** [comment]
Unfortunately, I don't know a good way to calculate Zmate. It is just taken from observations. We could allow the user to change it, but I don't think we can calculate it.

**Comment by 404Wolf** [comment]
@DNAcer I think I've arrived at a potential formula. [Here it is.](https://www.desmos.com/calculator/wwpzgmcnmf). I'm not sure why the modifier of 2 is needed. I think the issue is that NATuG is lacking the ability to input g.

**Comment by 404Wolf** [comment]
Update: I've deployed my formula for Z_mate computations to /main and everything seems to be working properly. If you get a chance to audit it, that'd be great. Thanks!

**Comment by 404Wolf** [comment]
Some parameter experimentation showing it working properly now.

![image](https://user-images.githubusercontent.com/108041238/204164808-a1d8aaa2-51af-4718-b894-c304abcb36b6.png)
![image](https://user-images.githubusercontent.com/108041238/204164823-d1248545-0ca4-4ca4-b55e-f89b6cf6e8e9.png)
![image](https://user-images.githubusercontent.com/108041238/204164854-4515d92d-85c4-4aac-ac68-7c8a921e00ec.png)
![image](https://user-images.githubusercontent.com/108041238/204164859-eb1b7bf4-c6a8-4739-b305-30d8cfcab547.png)

**Comment by DNAcer** [comment]
@404Wolf
You said you couldn't explain the factor of 2 in your formula, but can you explain any of it? Does it have any rationale other than that it gets the right answers in the cases you've tested?

**Comment by DNAcer** [comment]
@404Wolf 
Currently the system does not deal with changes to B and T remotely correctly. If you try changing to 1 turn every 21 bases, it rescales the graph, but still has 2 turns every 21 bases. If you try 1 turn every 10 bases, it puts 10 bases in each turn, but it does not allocate the bases correctly. In fact, when I change the values of T and B, the top view graph does not change at all, even as I update graphs and the side view changes.

**Comment by DNAcer** [comment]
@404Wolf I'm leaving this here because I assume it's related to the work you've been doing on this topic: the system is mucking up connections from the left and right edges of the tube. See attached image.
<img width="1072" alt="Screen Shot 2022-11-28 at 11 26 35 AM" src="https://user-images.githubusercontent.com/118033291/204330030-d109bf7c-a3dd-464f-a7ac-bf161d4885dd.png">

**Comment by 404Wolf** [comment]
> @404Wolf You said you couldn't explain the factor of 2 in your formula, but can you explain any of it? Does it have any rationale other than that it gets the right answers in the cases you've tested?

g is the angle between a given NEMid and its mate on the other helix. 
g/360 is the fraction of a full turn between a NEMid and its mate on the other helix.
The next step would be to somehow convert this fraction of a twist into an actual height, and this is where I got a bit confused. (g/360)H is 1.3nm, which is far off from .94. So you're right, I chose to multiply by Z_s without much understanding of why it makes the numbers work. 
Even if this exact formula isn't correct, I do think that g was what we were missing to be able to calculate a formula. From my understanding of the glossery, g is the same as theta_mate. I'm not 100% sure how to convert g to a height.
